twill-yarn/src/main/java/org/apache/twill/internal/appmaster/RunningContainers.java

```diff
@@ -475,6 +496,11 @@ private void sendMessage(final String runnableName, final Message message,
     Futures.addCallback(controller.sendMessage(message), new FutureCallback<Message>() {
       @Override
       public void onSuccess(Message result) {
+        //update log level information
```

Comment:
onSuccess here means that the message has been processed by the TwillContainerService, and the log levels have been set up. If there is some error setting up the log levels, an failed future with exception will be returned to callback, so it will go to onFailure() method